tests/lib/rule-stub.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @fileoverview Tests for stub rules.
+ * @author Ian VanSchooten
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var assert = require("chai").assert,
+    sinon = require("sinon"),
+    rulesStub = require("../../lib/rule-stub");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+describe("rule-stub", function() {
+    var sandbox;
+
+    beforeEach(function() {
+        sandbox = sinon.sandbox.create();
+    });
+
+    afterEach(function() {
+        sandbox.verifyAndRestore();
+    });
+
+    it("should throw an error when not given a warning", function() {
+        assert.throws(function() {
+            rulesStub();
+        }, "No message passed to rule stub");
+    });
+
+    it("should accept an error message", function() {
+        assert.doesNotThrow(function() {
```

Comment:
Since the function returns a function, you can call it with a fake context object and ensure it calls context.report when Program() is called.